### ISSUE TITLE
Path variable not correctly escaped in link

### ISSUE THREAD

**Issue by jimmythomson** [thymeleaf/thymeleaf#470]
I'm using Thymeleaf (3.0.0.BETA02) in my Spring MVC app.

I'm trying to create a URL. The code looks like this:

`<a href="edit.html" th:href="@{/slugs/edit/{id}(id=${slug.id})}">[[${slug.id}]]</a>`

This works fine, except that the value of ${slug.id} might be something like slug:/path/to/page (the ID is used for storage in a Couchbase repository), and this value is NOT escaped. So I end up with a URL that looks like this: /slugs/edit/slug:/path/to/page

This then gives me a 404 because the URL doesn't map to a controller.

What I want is to have the URL created so that it's escaped:

/slugs/edit/slug%3A%2Fpath%2Fto%2Fpage

I can see that in the StandardLinkBuilder class, at line 344, the 'escapedReplacement' variable is defined and assigned to. This only gets run through 'UriEscape.escapeUriPath' when there is actually a question mark added at the end of the url string. If it's not there then 'start' is -1 and 'UriEscape.escapeUriPath' never gets executed.

The other issue (this may be by design, I'm not sure), but I think that the use of 'UriEscape.escapeUriPath' should be 'escapeUriPathSegment' instead.

**Comment by danielfernandez** [comment]
OK, so two things here:
1. It seems path escaping is not being applied where there is no `?`. I wonder how this passed the tests...
2. The _escapeUriPath_/_escapeUriPathSegment_ thing: when this was implemented, it was implemented to mimic the behaviour of Spring's URI template patterns in Spring's JSP `<s:url>` tags. Back then, Spring's `UrlTag` class behaviour was to only use _path escaping_, but it seems in Spring 4.1 a possibility was added to do _path segment escaping_ if the variable was specified like `{/slug.id}` (note the slash). So this could be a nice addition to the syntax. I've just added it here: #471
